src/layout/layout.js

```diff
@@ -284,6 +284,11 @@
       this.element_.parentElement.removeChild(this.element_);
       container.appendChild(this.element_);
 
+      var focusedElement = this.element_.querySelector('[autofocus]');
```

Comment:
The selector should be a general `:focus` selector. At this point focus should be given to anything with autofocus or could have been given by the user to some other element.